pkg/ipcache/ipcache.go

```diff
@@ -198,34 +217,36 @@ func ipIdentityWatcher(owner IPIdentityMappingOwner) {
 		// Get events from channel as they come in.
 		for event := range watcher.Events {
 
+			scopedLog := log.WithFields(logrus.Fields{"kvstore-event": event.Typ.String(), "key": event.Key})
+			scopedLog.Debug("received event")
+
 			var (
 				cacheChanged      bool
 				cacheModification CacheModification
 				ipIDPair          identity.IPIdentityPair
+				cachedIdentity    identity.NumericIdentity
 			)
 
-			// Key and value are empty for ListDone event types, so just continue.
-			// See GH-3159.
+			// Key and value are empty for ListDone event types; do not try
+			// to unmarshal key or value.
```

Comment:
Could this be a case in the switch below? I realise it was separate before but the unmarshal between the two is now somewhere else.

Reply:
Good point, this is cleaner.